src/vs/workbench/api/common/extHostTerminalService.ts

```diff
@@ -456,14 +456,18 @@ export abstract class BaseExtHostTerminalService implements IExtHostTerminalServ
 		// Processes should be initialized here for normal virtual process terminals, however for
 		// tasks they are responsible for attaching the virtual process to a terminal so this
 		// function may be called before tasks is able to attach to the terminal.
-		let retries = 5;
+		let retries = 8;
+		let currentTimeout = 50;
 		while (retries-- > 0) {
 			if (this._terminalProcesses[id]) {
```

Comment:
Unfortunately tweaking the timeout to just go longer could still cause the issue on slower machines.

Could we instead just move this to the bottom of `_setupExtHostProcessListeners`? We would want to make sure this is guaranteed to happen after `openPromise` is fired.

Alternatively we could add promises for `_terminalProcesses` like we do for `_getTerminalPromises` (`ExtHostTerminal`) and just rely on the promise instead of polling. This would probably be the better approach.

Reply:
Ok. Understood. My thought was that intent was to say that it's an error if we aren't able to establish a terminal, and during refactoring throwing an exception was forgotten.

Reply:
It would be but it shouldn't ever happen. If we move to the promise-based approach we don't need to do any polling and if for some reason the connection doesn't get made then you will have a broken terminal. That's far better than valid cases where it just took a long time, it will also happen as soon as possible if we use promises rather than waiting for 50ms+ every check.

Reply:
@Tyriar BTW, in _createGetTerminalPromise you have 1s to get terminal, otherwise it returns undefined. Is it normal? For me, it looks like the same error. What's worse undefined value is ignored in most of the cases. Using remoting will lead to arbitrarily long delays due to TCP/IP and all this code will break down in these cases.

Reply:
I mean this code:

```
	private _createGetTerminalPromise(id: number, retries: number = 5): Promise<ExtHostTerminal> {
		return new Promise(c => {
			if (retries === 0) {
				c(undefined);
				return;
			}

			const terminal = this._getTerminalById(id);
			if (terminal) {
				c(terminal);
			} else {
				// This should only be needed immediately after createTerminalRenderer is called as
				// the ExtHostTerminal has not yet been iniitalized
				timeout(200).then(() => c(this._createGetTerminalPromise(id, retries - 1)));
			}
		});
	}
```

Reply:
Yeah that's a similar problem we should fix, not sure if there's an issue for that one. There's #77655 which is tracking one of the issues that I fixed but needed to get reverted because it broke some tests.

Reply:
@Tyriar I followed a simple solution. I will try to fix the issues with a different PR. We will definitely experience the problems with longer delays. Will create a separate PR for this.

Reply:
@Tyriar So, the tests failed. Do you have any tips on what to do?

Reply:
@Tyriar Do you have any instructions on how to debug them? Or at least run only subset of them?